ui/app/js/controllers/user_edit_controller.js

```diff
@@ -5,6 +5,88 @@ function ($scope, $modalInstance, CSRF, Permissions, users, is_edit, user) {
   $scope.loading = true;
   $scope.users = users;
   $scope.currentItemTab = 1;
+
+  var allPermissions = {
+        "admin": ["products"],
+        "release": ["actions", "products"],
+        "release_locale": ["actions", "products"],
+        "release_read_only": ["actions", "products"],
+        "rule": ["actions", "products"],
+        "permission": ["actions"],
+        "required_signoff": ["products"],
+        "scheduled_change": ["actions"],
+    };
+
+  $scope.onPermission = function(){
+      $scope.available_options = allPermissions[$scope.permission.permission];
+
+      // clear  all previous selections 
+      [
+          $scope.selected_options,
+          $scope.selected_actions,
+          $scope.selected_products
+      ] = [[], [], []];
+  };
+  $scope.selected_options = [];
+
+  $scope.actions = ["create", "modify", "enact"];
+  $scope.selected_actions = [];
+  $scope.action = null;
+
+  $scope.products = [];
+  $scope.selected_products = [];
+  $scope.product = null;
+
+  // get products in releases
+  Permissions.getProducts().success(function(response) {
+    $scope.products = response.product;
+  });
+
+  $scope.$watch(function(){
+      // set actions & products to [] when undefined, always happens when running tests
+      $scope.actions = (typeof $scope.actions === 'undefined')? []: $scope.actions;
+      $scope.products= (typeof $scope.products === 'undefined')? []: $scope.products;
```

Comment:
When would they be set to undefined? They're initialized just a few lines up, so this should never be the case, even in tests....